### PR TITLE
Merge pull request #1444 from prisma/Fix-#1362

### LINKED ISSUE #1362: Improve formatting of missing generator error message
It's hard to tell the "useful error message with instructions" apart from the stacktrace. We should improve & streamline our CLI error formatting.

image

### PR DESCRIPTION
**Pull request by Jolg42**

### CODE REVIEW

File: cli/prisma2/src/bin.ts [hunk 1]
```diff
@@ -120,21 +120,38 @@ if (require.main === module) {
         process.exit(code)
       }
     })
-    .catch(err => {
-      if (err.rustStack) {
-        handlePanic(err, packageJson.name, packageJson.version).catch(e => {
+    .catch((err) => {
+      let shouldExit = false
+
+      function handleIndividualError(err) {
```
Comment (timsuchanek) [inline cli/prisma2/src/bin.ts]: This introduces **shadowing** of the variable `err`. Please name it differently in the `handleIndividualError` method.

File: cli/prisma2/src/bin.ts [hunk 2]
```diff
@@ -120,21 +120,38 @@ if (require.main === module) {
...
+      // Sindre's pkg p-map & co are using AggregateError, it is an iterator.
+      if (typeof err[Symbol.iterator] === 'function') {
+        for (const individualError of err) {
+          handleIndividualError(individualError)
+        }
+      } else {
+        handleIndividualError(err)
+      }
+
+      if (shouldExit) {
```
Comment (timsuchanek) [inline cli/prisma2/src/bin.ts]: Why do we need `shouldExit`? Shouldn't it always exit?

**Review by timsuchanek — CHANGES_REQUESTED** [review]

File: cli/prisma2/src/bin.ts (hunk 2 above)
Comment (Jolg42) [reply cli/prisma2/src/bin.ts]: So this logic was here before, my understanding is that when we have a err.rustPanic we delegate this to this custom interactive logic?

File: cli/prisma2/src/bin.ts (hunk 1 above)
Comment (Jolg42) [reply cli/prisma2/src/bin.ts]: Indeed! Will fix asap :)

**Pull request merged**